DESIGN_GUIDELINES.md

```diff
@@ -1165,6 +1225,58 @@ use the SecretValue type [_awslint:secret-token_].
 * No need to put every class in a separate file. Try to think of a
   reader-friendly organization of your source files.
 
+### Levels of Abstraction
+
+There are three different levels of constructs in this library, beginning with
+low-level constructs, which we call _CFN Resources_ (or L1, short for "level 1")
+or Cfn (short for CloudFormation) resources. These constructs directly represent
+all resources available in AWS CloudFormation. CFN Resources are periodically
+generated from the AWS CloudFormation Resource Specification. They are named
+**Cfn**_Xyz_, where _Xyz_ is name of the resource. For example, CfnBucket
+represents the AWS::S3::Bucket AWS CloudFormation resource. When you use Cfn
+resources, you must explicitly configure all resource properties, which requires
+a complete understanding of the details of the underlying AWS CloudFormation
+resource model.
+
+The next level of constructs, L2, also represent AWS resources, but with a
+higher-level, intent-based API. They provide similar functionality, but provide
+the defaults, boilerplate, and glue logic you'd be writing yourself with a CFN
+Resource construct. AWS constructs offer convenient defaults and reduce the need
+to know all the details about the AWS resources they represent, while providing
+convenience methods that make it simpler to work with the resource. For example,
+the s3.Bucket class represents an Amazon S3 bucket with additional properties
+and methods, such as bucket.addLifeCycleRule(), which adds a lifecycle rule to
+the bucket.
+
+Examples of behaviors that an L2 commonly include:
+
+- Modeling of the underlying L1 properties
+- Methods for adding related resources (e.g., adding an event notification to
+  an S3 bucket).
+- Modeling of permissions and resource policies
+- Modeling of metrics
+
+In addition to the above, some higher-order L2s may introduce more complex and
+helpful functionality, either part of the original L2 itself, or as part of a
+separate construct. The most common form of these L2s are integration constructs
+that model interactions between different services (e.g., SNS publishing to SQS,
+CodePipeline actions that trigger Lambda functions).
+
+The final level of abstraction present within the CDK are what we designate as
```

Comment:
At the moment, we have ecs-patterns and route53-patterns as "part of the cdk". The consumption model is exactly like consuming an L2 or L2.5.

It seems to me that this is applicable only after we do CDKv2. Should we be more clear on this?

Reply:
Added.